course-sections/section-04-if/assets/scripts/app.js

```diff
@@ -0,0 +1,98 @@
+const defaultResult = 0;
+let currentResult = defaultResult;
+let logEntries = [];
+// also works 
+// let logEntries = [1, 2, 3];
+//Just one more comment!
+
+function getUserNumberInput() {
+    return parseInt(usrInput.value);
+}
+
+// Generates and writes calculation log
+
+function writeToLog(
+    operationIdentifier,
+    prevResult,
+    operationNumber,
+    newResult
+){
+    const logEntry = {
+        operation: operationIdentifier,
+        prevResult: prevResult,
+        number: operationNumber,
+        result: newResult
+    };
+    logEntries.push(logEntry);
+    console.log(logEntry.operation);
+    console.log(logEntries);
+}
+
+function createAndWriteOutput (operator, resultBeforeCalc,calcNumber){
+    const calcDescription = `${resultBeforeCalc} ${operator} ${calcNumber}`;
+    outputResult(currentResult, calcDescription);// from vendor file
+}
+
+function calculatedResult (calculationType) {
+  if (
+    calculationType !== "ADD" &&
+    calculationType !== "SUBTRACT" &&
+    calculationType !== "MULTIPLY" &&
+    calculationType !== "DIVIDE"
+    !enteredNumber
+  ) {
+    return;
```

Comment:
Nice. It called "defensive programming". You are checking for correct argument.

However, you must not only "return". You must also throw an ERROR (like `console.error`)